src/main/java/bwapi/BWClientConfiguration.java

```diff
@@ -0,0 +1,77 @@
+package bwapi;
+
+/**
+ * Configuration for constructing a BWClient
+ */
+public class BWClientConfiguration {
```

Comment:
I'd prefer the variables to be private and modifiable using a `withDebugConnection()` , with the comment above the variable added to the `withX` method

example API in my mind:
```Java
BWClientConfiguration bwc = new BWClientConfiguration()
    .withAutoContinue()
    .withAsync()
    .withMaxFrameDuration(30);
```

```Java
/**Javadoc*/
public BWClientConfiguration withMaxFrameDuration(int value) {
    if (value < 1) {
        throw new IllegalArgumentException("maxFrameDuration needs to be a non-negative number (it's how long JBWAPI waits for a bot response before returning control to BWAPI).");
    }
    this.maxFrameDuration = value;
    return this;
}
```

This way we don't need a "validate" method as we can check the arguments on creating the configuration
and users don't need to change internal variable names of objects.

Reply:
Will do, although I think the validate method is still necessary since it considers interactions between properties and it would be annoying to require them to be set in a specific order, and it doesn't prevent modification of the properties.